plugins/modules/aci_dhcp_relay_provider.py

```diff
@@ -0,0 +1,357 @@
+     returned: failure or debug
+     type: str
+     sample: OK (30 bytes)
+   status:
+     description: The HTTP status from the APIC
+     returned: failure or debug
+     type: int
+     sample: 200
+   url:
+     description: The HTTP url used for the request to the APIC
+     returned: failure or debug
+     type: str
+     sample: https://10.11.12.13/api/mo/uni/tn-production.json
+   '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        relay_policy=dict(type='str', aliases=['relay_policy_name']),
+        epg_type=dict(type='str', required=True,
+                      choices=['app_epg',
+                               'l2_external_net',
+                               'l3_external_net']),
+        anp=dict(type='str'),
+        app_epg=dict(type='str'),
+        l2out_name=dict(type='str'),
+        l3out_name=dict(type='str'),
+        external_net=dict(type='str'),
+        dhcp_server_addr=dict(type='str'),
+        state=dict(type='str', default='present',
+                   choices=['absent', 'present', 'query']),
+        tenant=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['relay_policy', 'tenant']],
+            ['state', 'present', ['relay_policy', 'tenant']],
+            ['epg_type', 'app_epg', ['anp', 'app_epg']],
+            ['epg_type', 'l2_external_net', ['l2out_name', 'external_net']],
+            ['epg_type', 'l3_external_net', ['l3out_name', 'external_net']],
+        ],
+        mutually_exclusive=[
+            ['anp', 'l2out_name'],
+            ['anp', 'l3out_name'],
+            ['anp', 'external_net'],
+            ['app_epg', 'l2out_name'],
+            ['app_epg', 'l3out_name'],
+            ['app_epg', 'external_net'],
+            ['l2out_name', 'l3out_name'],
+        ],
+    )
+
+    relay_policy = module.params.get('relay_policy')
+    state = module.params.get('state')
+    tenant = module.params.get('tenant')
+    epg_type = module.params.get('epg_type')
+    anp = module.params.get('anp')
+    app_epg = module.params.get('app_epg')
+    l2out_name = module.params.get('l2out_name')
+    l3out_name = module.params.get('l3out_name')
+    external_net = module.params.get('external_net')
+    dhcp_server_addr = module.params.get('dhcp_server_addr')
+
+    if epg_type == 'app_epg':
+        tdn = 'uni/tn-{0}/ap-{1}/epg-{2}'.format(tenant, anp, app_epg)
+    elif epg_type == 'l2_external_net':
+        tdn = ('uni/tn-{0}/l2out-{1}/instP-{2}'
+               .format(tenant, l2out_name, external_net))
+    elif epg_type == 'l3_external_net':
+        tdn = ('uni/tn-{0}/out-{1}/instP-{2}'
+               .format(tenant, l3out_name, external_net))
```

Comment:
tenant here can also be provider_tenant as I mentioned above so we should have a separate field for that. We can consider provider_tenant = tenant if not provided for example.

Reply:
Added.